prow/cmd/build/controller_test.go

```diff
@@ -161,6 +182,62 @@ func (fl *fakeLimiter) NumRequeues(item interface{}) int {
 	return 0
 }
 
+type fakeNsLister struct {
+	pjs map[string]prowjobv1.ProwJob
+}
+
+func newFakeNsLister(pjs map[string]prowjobv1.ProwJob) *fakeNsLister {
+	return &fakeNsLister{
+		pjs: pjs,
+	}
+}
+
+func (f *fakeNsLister) List(selector labels.Selector) ([]*prowjobv1.ProwJob, error) {
+	result := []*prowjobv1.ProwJob{}
+	for _, pj := range f.pjs {
+		newPj := pj
```

Comment:
This is tricky and I can't remember what it does with the underlying data -- can you do

```go
for i := range f.pjs {
    result = append(result, f.pjs[i].DeepCopy())
}
```

Reply:
Switched to DeepCopy